src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -74,7 +77,7 @@ public class Session {
             _pipe = new MessagePipe(messageLogger);
         }
 
-        public void StartHost(SecureString password, string profilePath, ILogger outputLogger, LogVerbosity verbosity) {
+        public void StartHost(ClaimsPrincipal cp, string profilePath, ILogger outputLogger, LogVerbosity verbosity) {
```

Comment:
If we're only using the principal to retrieve the password, is it worth passing the whole thing? This way we get the non-secure string flow through and persist around at least until this method returns...